### PR TITLE
Update apis for taxonomies

### PR DESCRIPTION
**Pull request by marcincichon**
@ollietreend can you review?

### CODE REVIEW

File: wp-content/themes/mojintranet/modules/api/widgets.php [hunk 1]
```diff
@@ -39,41 +39,46 @@ protected function parse_params($params) {
...
+    $options = $this->add_taxonomies($options);
+    $data = $this->MVC->model->need_to_know->get_data($options);
     $data['url_params'] = $this->params;
     $this->response($data, 200, 60);
   }
 
   private function get_my_moj() {
-    $data = $this->MVC->model->my_moj->get_data($this->params['agency']);
+    $options = $this->params;
+    $options = $this->add_taxonomies($options);
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/modules/api/widgets.php]: Is this required here? It doesn't look like `$options['tax_query']` is used by the `my_moj` model.

File: wp-content/themes/mojintranet/modules/api/widgets.php [hunk 2]
```diff
@@ -39,41 +39,46 @@ protected function parse_params($params) {
...
+    $data = $this->MVC->model->my_moj->get_data($options);
     $data['url_params'] = $this->params;
     $this->response($data, 200, 60 * 60);
   }
 
   private function get_follow_us_links() {
     $this->MVC->model('follow_us');
-    $data = $this->MVC->model->follow_us->get_data($this->params['agency']);
+    $options = $this->params;
+    $options = $this->add_taxonomies($options);
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/modules/api/widgets.php]: Same thing here – is `add_taxonomies` required?

File: wp-content/themes/mojintranet/modules/api/widgets.php [hunk 3]
```diff
@@ -39,41 +39,46 @@ protected function parse_params($params) {
...
     $data = $this->MVC->model->news->get_widget_news($options, $featured);
     $data['url_params'] = $this->params;
     $this->response($data, 200, 60);
   }
 
   private function get_need_to_know() {
-    $data = $this->MVC->model->need_to_know->get_need_to_know($this->params);
+    $options = $this->params;
+    $options = $this->add_taxonomies($options);
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/modules/api/widgets.php]: Same here – is `add_taxonomies` required?

File: wp-content/themes/mojintranet/models/my_moj.php [hunk 4]
```diff
@@ -1,7 +1,10 @@
 <?php if (!defined('ABSPATH')) die();
 
 class My_moj_model extends MVC_model {
-  function get_data($agency = 'hq') {
+  function get_data($options) {
+    $agency = $options['agency'] ?: 'hq';
+    $additional_filters = $options['additional_filters'] ?: '';
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/models/my_moj.php]: `$additional_filters` isn't used in this method – could it be removed?

File: wp-content/themes/mojintranet/models/follow_us.php [hunk 5]
```diff
@@ -1,7 +1,10 @@
 <?php
 
 class Follow_us_model extends MVC_model {
-  function get_data($agency = 'hq') {
+  function get_data($options) {
+    $agency = $options['agency'] ?: 'hq';
+    $additional_filters = $options['additional_filters'] ?: '';
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/models/follow_us.php]: `$additional_filters` isn't used in this method – could it be removed?

Comment (marcincichon) [inline wp-content/themes/mojintranet/models/follow_us.php]: Yeah I kind of added that to every model just for consistency. You can't use it with the API as the url segments are defined but you can actually use it with the model directly. Again, not needed at the mo, but still. What do you think?

File: wp-content/themes/mojintranet/models/my_moj.php (hunk 4 above)
Comment (marcincichon) [inline wp-content/themes/mojintranet/models/my_moj.php]: ditto

File: wp-content/themes/mojintranet/modules/api/widgets.php (hunk 3 above)
Comment (marcincichon) [inline wp-content/themes/mojintranet/modules/api/widgets.php]: Yes, these are actually required for the widget to be filtered by agency.

File: wp-content/themes/mojintranet/modules/api/widgets.php (hunk 1 above)
Comment (marcincichon) [inline wp-content/themes/mojintranet/modules/api/widgets.php]: I think you're right - this one is redundant

File: wp-content/themes/mojintranet/modules/api/widgets.php (hunk 2 above)
Comment (marcincichon) [inline wp-content/themes/mojintranet/modules/api/widgets.php]: True, that's also redundant

**Comment by ollietreend** [comment]
Looks good @marcincichon

**Pull request merged**